inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/DefaultMessageSender.java

```diff
@@ -191,11 +194,39 @@ public String getSDKVersion() {
         return ConfigConstants.PROXY_SDK_VERSION;
     }
 
+    private SendResult retryWhenSendMessageFail(Function<Sender, SendResult> sendOperation) {
+        int attempts = 0;
+        SendResult sendResult = null;
+        while (attempts < this.senderMaxRetry + 1) {
```

Comment:
No need to increase by 1, senderMaxRetry has defined the maximum number of retries. If you increase by 1 during comparison, it will send one more time than the set value.

Reply:
I think the retryTime here should be the number of retries rather than the total number of executions, and by default it should be executed once like the original code, so I add 1 here

Reply:
done